servers/visual/rasterizer.h

```diff
@@ -969,6 +969,54 @@ class RasterizerCanvas {
 						for (int j = 1; j < l; j++) {
 							r.expand_to(pp[j]);
 						}
+
+						if (skeleton != RID()) {
+							// calculate bone AABBs
+							int bone_count = RasterizerStorage::base_singleton->skeleton_get_bone_count(skeleton);
+
+							Vector<Rect2> bone_aabbs;
+							bone_aabbs.resize(bone_count);
+
+							for (int j = 0; j < bone_count; j++) {
+								bone_aabbs.write[j].size = Vector2(-1, -1); //negative means unused
+							}
+
+							Rect2 *bptr = bone_aabbs.ptrw();
```

Comment:
```suggestion
							Vector<Rect2> bone_aabbs;
							bone_aabbs.resize(bone_count);
							Rect2 *bptr = bone_aabbs.ptrw();

							for (int j = 0; j < bone_count; j++) {
								bptr[j].size = Vector2(-1, -1); //negative means unused
							}
```

May help with performance a bit I think. In any case the code below modifies the `bone_aabbs` through `bptr` the same way as with suggested changes, without using the `write` proxy. 🙂

Reply:
Good thinking! Thanks!